[Feature-8386][Server] Custom timezone - Add timezone into user info

jim-parsons opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request

Custom timezone - Add timezone into user info

<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log


  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Add system default timezone like `GMT+08:00` when create user
  3. Add timeZone property when update user info

<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Add system default timezone like `GMT+08:00` when create user
  3. Add timeZone property when update user info

This completes #8386

jim-parsons committed:
Merge remote-tracking branch 'upstream/dev' into Feature-8386

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Add system default timezone like `GMT+08:00` when create user
  3. Add timeZone property when update user info
  4. add ddl sql in update sql script

This completes #8386

jim-parsons commented:
@caishunfeng plz review，thx~

caishunfeng commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:
```diff
@@ -205,6 +207,8 @@ public User createUser(String userName,
         }
         user.setQueue(queue);
+        String defaultTimeZone = ZoneId.systemDefault().getRules().getOffset(Instant.now()).getId();
+        user.setTimeZone("GMT" + defaultTimeZone);
```
Is it better to init the user timezone null? If timezone null, use the server default timezone for date convert. What do you think?

caishunfeng commented on dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UsersServiceTest.java:
```diff
@@ -269,13 +269,13 @@ public void testUpdateUser() {
         try {
             //user not exist
-            Map<String, Object> result = usersService.updateUser(getLoginUser(), 0, userName, userPassword, "3443@qq.com", 1, "13457864543", "queue", 1);
+            Map<String, Object> result = usersService.updateUser(getLoginUser(), 0, userName, userPassword, "3443@qq.com", 1, "13457864543", "queue", 1, "+8");
```
```suggestion
            Map<String, Object> result = usersService.updateUser(getLoginUser(), 0, userName, userPassword, "3443@qq.com", 1, "13457864543", "queue", 1, "Asia/Shanghai");
```

caishunfeng commented on dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UsersServiceTest.java:
```diff
@@ -269,13 +269,13 @@ public void testUpdateUser() {
             //success
             when(userMapper.selectById(1)).thenReturn(getUser());
-            result = usersService.updateUser(getLoginUser(), 1, userName, userPassword, "32222s@qq.com", 1, "13457864543", "queue", 1);
+            result = usersService.updateUser(getLoginUser(), 1, userName, userPassword, "32222s@qq.com", 1, "13457864543", "queue", 1, "+8");
```
```suggestion
            result = usersService.updateUser(getLoginUser(), 1, userName, userPassword, "32222s@qq.com", 1, "13457864543", "queue", 1, "Asia/Shanghai");
```

caishunfeng commented on dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_h2.sql:
```diff
@@ -979,7 +980,7 @@ VALUES ('1,2', 1, 'default admin warning group', 'default admin warning group',
 INSERT INTO t_ds_user
 VALUES ('1', 'admin', '7ad2410b2f4c074479a8937a28a22b8f', '0', 'xxx@qq.com', '', '0', '2018-03-27 15:48:50',
-        '2018-10-24 17:40:22', null, 1);
+        '2018-10-24 17:40:22', null, 1, 'GMT+8');
```
```suggestion
        '2018-10-24 17:40:22', null, 1, null);
```

caishunfeng commented on dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.4_schema/mysql/dolphinscheduler_ddl.sql:
```diff
@@ -14,3 +14,5 @@
  * limitations under the License.
 */
+
+ALTER TABLE `t_ds_user` ADD COLUMN `time_zone` varchar(4) COMMENT 'time zone, e.g. +8' AFTER `state`;
```
```suggestion
ALTER TABLE `t_ds_user` ADD COLUMN `time_zone` varchar(4) COMMENT 'time zone' AFTER `state`;
```

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info

 1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

jim-parsons committed:
Merge remote-tracking branch 'upstream/dev' into Feature-8386

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info

1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info

1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

jim-parsons commented:
@caishunfeng i‘ve updated the code, thx for your advice, plz take a review

caishunfeng reviewed (approved):
LGTM

caishunfeng commented:
> @caishunfeng i‘ve updated the code, thx for your advice, plz take a review

Look good to me, I had merged the pr, thanks for your contribution.

caishunfeng commented:
this pr close #8386
